inspector/main.py

```diff
@@ -271,3 +281,99 @@ def health():
 @app.route("/robots.txt")
 def robots():
     return Response("User-agent: *\nDisallow: /", mimetype="text/plain")
+
+
+@app.route(
+    "/project/<project_name>/<version>/packages/<first>/<second>/<rest>/<distname>/download"
+)
+def download_distribution(project_name, version, first, second, rest, distname):
+    """Download entire distribution package."""
+    if project_name != canonicalize_name(project_name):
+        return redirect(
+            url_for(
+                "download_distribution",
+                project_name=canonicalize_name(project_name),
+                version=version,
+                first=first,
+                second=second,
+                rest=rest,
+                distname=distname,
+            ),
+            301,
+        )
+
+    try:
+        dist = _get_dist(first, second, rest, distname)
+    except InspectorError:
+        return abort(400)
+
+    if not dist:
+        return abort(404)
+
+    # Get the raw distribution file from PyPI
+    url = f"https://files.pythonhosted.org/packages/{first}/{second}/{rest}/{distname}"
+    try:
+        resp = requests_session().get(url, stream=True)
+        resp.raise_for_status()
+    except Exception:
+        return abort(404)
+
+    # Return the file with proper headers to trigger download
+    return Response(
+        resp.content,
+        mimetype="application/octet-stream",
+        headers={
+            "Content-Disposition": f"attachment; filename={distname}",
+            "Content-Length": str(len(resp.content)),
+        },
+    )
```

Comment:
I think we can just provide a direct link to the URL where the distribution is hosted on PyPI here, rather than passing it through the application.